lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
-  });
+  const quickRender = rtlRender(NxDropdown, minimalProps);
+  const renderEl = rtlRenderElement(NxDropdown, minimalProps);
 
   it('renders the button according to the supplied variant', function() {
-    let component = getMountedComponent();
-    expect(component.find(NxButton)).toHaveProp('variant', 'tertiary');
+    const { rerender } = quickRender();
+
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--tertiary');
 
-    component = getMountedComponent({ variant: 'primary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'primary');
+    rerender(<NxDropdown {...minimalProps} variant="primary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--primary');
 
-    component = getMountedComponent({ variant: 'secondary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'secondary');
+    rerender(<NxDropdown {...minimalProps} variant="secondary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--secondary');
 
-    component = getMountedComponent({ variant: 'error' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'error');
+    rerender(<NxDropdown {...minimalProps} variant="error" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--error');
   });
 
   it('correctly renders the menu based on isOpen prop', function() {
-    let component = getMountedComponent({ isOpen: true });
-    expect(component.find(NxDropdownMenu)).toExist();
+    const { container, rerender } = quickRender({ ...minimalProps, isOpen: true });
+    const menu = container.querySelector('.nx-dropdown-menu');
 
-    component = getMountedComponent({ isOpen: false });
-    expect(component.find(NxDropdownMenu)).not.toExist();
+    expect(menu).toBeInTheDocument();
+    rerender(<NxDropdown {...minimalProps} isOpen={false} />);
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('correctly assigns supplied classes', function() {
-    const component = getShallowComponent({ className: 'class1 class2' });
-    expect(component).toHaveClassName('nx-dropdown class1 class2');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('disables the button (and the toggle fn) when the disabled prop is supplied', function() {
-    const toggleFn = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse: toggleFn, disabled: true }),
-        button = component.find(NxButton);
+  it('disables the button (and the toggle fn) when the disabled prop is supplied', async function() {
+    const user = userEvent.setup();
+    const toggleFn = jest.fn();
+    renderEl({ onToggleCollapse: toggleFn, disabled: true });
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('disabled');
```

Comment:
The classname is an impl detail and doesn't fully convey the a11y metadata we want to test
```suggestion
    expect(button).toBeDisabled();
 ```

Reply:
> The classname is an impl detail and doesn't fully convey the a11y metadata we want to test

Hmm this failed for me because it doesn't have the disabled attribute.

Reply:
> Hmm this failed for me because it doesn't have the disabled attribute.

Then that attribute should be added (or at least `aria-disabled` if the actual `disabled` attribute is a problem for some reason)

Reply:
> > Hmm this failed for me because it doesn't have the disabled attribute.
> 
> Then that attribute should be added (or at least `aria-disabled` if the actual `disabled` attribute is a problem for some reason)

Added that here: 260fe8b412cb9a7cd78cdc45abb36d00685ebfc0